base/strings/util.jl

```diff
@@ -575,6 +580,11 @@ function bytes2hex(a::AbstractArray{UInt8})
     return String(b)
 end
 
+bytes2hex(io::IO, a::AbstractArray{UInt8}) =
+    for x in a
+        print(io, Char(hex_chars[1 + x >> 4]), Char(hex_chars[1 + x & 0xf]))
```

Comment:
`write(io, hex_chars[1 + (x >> 4)])` would probably be faster, since it'll save us from needing to decode the Char into the stream

Reply:
Faster but wrong, unfortunately.  Writing bytes directly assumes that `io` uses an ASCII-compatible encoding (#26286).   We would need an `ASCIIChar` type to get both encoding-independence and maximum performance here.